### PR TITLE
Netbox_device.py module

### PR DESCRIPTION
**Pull request by FragmentedPacket**
##### SUMMARY

netbox_device module to create/delete devices in Netbox
Related to #45948 

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

netbox_device

##### ADDITIONAL INFORMATION

**netbox_create_device():**
This function will normalize the data, find any keys that require the IDs of the objects and then attempt to create the device within Netbox

**normalize_data():**
This function will convert the data to JSON and then attempt to normalize any of the values in the data that will be used to find the ID of the object (uses the QUERY_TYPES dictionary to determine how this data should look)

**find_ids():**
This function iterates through the keys in the data provided and checks them against the CONVERT_TO_ID dictionary. If the key is within the dictionary, it will then go through the process of building the Pynetbox object to get the ID. If it comes back with data, it will use the ID it found, if not, it will then check to see if the key is in the NO_DEFAULT_ID dictionary to make sure it keeps the data as is else it will give a default value of 1.

**find_app():**
This function is kicked off currently within the find_ids() method. This helps return the application that we need for any endpoint that may be in the data that we need to query for.

**netbox_delete_device():**
This uses the normalize_data function and then attempts to retrieve the object using pynetbox and a name(this returns either an object or None). If it gets an object back, it will then attempt to delete the device.

### CODE REVIEW

**Comment by FragmentedPacket** [comment]
@sieben @Anthony25 @nikkytub @amb1s1 @pilou- 

Hey guys,

I'm not sure if y'all want to be involved in the maintenance of the Netbox modules I'm in the process of creating, but let me know and I can add these modules to the BOTMETA. If not, I'd appreciate some feedback regardless :)

**Comment by FragmentedPacket** [comment]
shipit

**Comment by FragmentedPacket** [comment]
@gundalow Hey John. Is there anything further that needs to be done to get this merged?

File: lib/ansible/modules/net_tools/netbox/netbox_device.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: netbox_device
+short_description: Manage devices
```
Comment (gundalow) [inline lib/ansible/modules/net_tools/netbox/netbox_device.py]: Needs to include the platform. This string is used on https://docs.ansible.com/ansible/latest/modules/list_of_all_modules.html

Comment (FragmentedPacket) [reply lib/ansible/modules/net_tools/netbox/netbox_device.py]: @gundalow We discussed this over slack and it has been updated. Thanks btw!

@amb1s1 also said he wouldn't mind being pinged on any changes so we'll keep him on there.

Is it fine if we merge this now?

**Review by gundalow — APPROVED** [review]

**Pull request merged**